extensions/mssql/src/sqlAssessment/sqlAssessmentService.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as mssql from '../mssql';
+import { AppContext } from '../appContext';
+import { SqlOpsDataClient, ISqlOpsFeature } from 'dataprotocol-client';
+import { ClientCapabilities } from 'vscode-languageclient';
+import * as constants from '../constants';
+import * as azdata from 'azdata';
+import * as contracts from '../contracts';
+
+
+export class SqlAssessmentService implements mssql.ISqlAssessmentService {
+	public static asFeature(context: AppContext): ISqlOpsFeature {
+		return class extends SqlAssessmentService {
+			constructor(client: SqlOpsDataClient) {
+				super(context, client);
+			}
+
+			fillClientCapabilities(capabilities: ClientCapabilities): void {
+			}
+
+			initialize(): void {
+			}
+		};
+	}
+
+	private constructor(context: AppContext, protected readonly client: SqlOpsDataClient) {
+		context.registerService(constants.SqlAssessmentService, this);
+	}
+	async assessmentInvoke(ownerUri: string, targetType: mssql.SqlAssessmentTargetType): Promise<mssql.SqlAssessmentResult> {
```

Comment:
nit: return `Promise<mssql.SqlAssessmentResult | undefined>`

(for this and the other functions in this file)